app/internal/notification.py

```diff
@@ -0,0 +1,174 @@
+from operator import attrgetter
+from typing import List, Union
+
+from fastapi import HTTPException
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from starlette.status import HTTP_401_UNAUTHORIZED, HTTP_406_NOT_ACCEPTABLE
+
+from app.database.models import (
+    Invitation,
+    Message,
+    InvitationStatusEnum,
+    MessageStatusEnum,
+)
+from app.internal.utils import create_model
+
+
+WRONG_NOTIFICATION_ID = (
+    "The notification id you have entered is wrong\n."
+    "If you did not enter the notification id manually, report this exception."
+)
+
+NOTIFICATION_TYPE = Union[Invitation, Message]
+
+UNREAD_STATUS = [
+    InvitationStatusEnum.UNREAD,
+    MessageStatusEnum.UNREAD,
+]
+
+ARCHIVED = [
```

Comment:
You check if things are `in` ARCHIVED`/`UNREAD_STATUS` - prefer `{` to create a `set`